pkg/kubectl/cmd/rollout/rollout_pause.go

```diff
@@ -89,32 +90,44 @@ func (o *PauseConfig) CompletePause(f *cmdutil.Factory, cmd *cobra.Command, out
 		return err
 	}
 
-	infos, err := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+	r := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
 		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, o.Recursive, o.Filenames...).
 		ResourceTypeOrNameArgs(true, args...).
-		SingleResourceType().
+		ContinueOnError().
 		Latest().
-		Do().Infos()
+		Flatten().
+		Do()
+	err = r.Err()
 	if err != nil {
 		return err
 	}
-	if len(infos) != 1 {
-		return fmt.Errorf("rollout pause is only supported on individual resources - %d resources were found", len(infos))
+
+	err = r.Visit(func(info *resource.Info, err error) error {
```

Comment:
Why cycle through like this instead of doing `.Infos()` above?  Does this do something different?

In other commands, this visitor let's you return errors and have them handled without failing the whole command, but here its just information gathering, right?